modules/kit-elements-defaults/assets/js/editor/commands/create.js

```diff
@@ -1,40 +1,66 @@
-import localSettingsExtractor from '../settings-handlers/local/extract';
-import loadDefaultValues from '../load-default-values';
+import store from '../store';
 
 export default class Create extends $e.modules.CommandBase {
-	settingsExtractors = [
-		localSettingsExtractor,
-	];
-
-	validateArgs( args = {} ) {
+	validateArgs() {
 		this.requireContainer();
 	}
 
 	async apply( { container } ) {
 		$e.internal( 'panel/state-loading' );
 
-		const type = container.model.get( 'widgetType' ) || container.model.get( 'elType' );
+		const elementType = container.model.get( 'widgetType' ) || container.model.get( 'elType' );
+
+		const settings = this.extractLocalSettings( container );
+
+		[ '__dynamic__', '__globals__' ].forEach( ( type ) => {
+			const specialSettings = this.extractSpecialSettings( type, container );
 
-		const settings = this.settingsExtractors.reduce( ( carry, extractor ) => {
-			return {
-				...carry,
-				...extractor( container ),
-			};
-		}, {} );
+			if ( ! Object.keys( specialSettings ).length ) {
+				return;
+			}
+
+			settings[ type ] = specialSettings;
+		} );
 
 		try {
-			await $e.data.create( 'kit-elements-defaults/index', { settings }, { type } );
+			await store.upsert( elementType, settings );
 
-			// Refresh Cache.
-			// Cannot use {refresh: true} in `get()` because in the hooks there must be a way to get the data
-			// in sychronous way, and when using `refresh: true`, the data will not be available in syncronous way.
-			loadDefaultValues();
+			elementor.notifications.showToast( {
+				message: __( 'Default values saved successfully. Please avoid saving sensetive data like passwords and api keys.', 'elementor' ),
+			} );
+		} catch ( error ) {
+			elementor.notifications.showToast( {
+				message: __( 'An error occurred.', 'elementor' ),
+			} );
 
-			// Here we should open some toast
-		} catch ( e ) {
-			// Show some error.
+			throw error;
 		} finally {
 			$e.internal( 'panel/state-ready' );
 		}
 	}
+
+	extractLocalSettings( container ) {
+		const controls = container.settings.controls,
+			settingsWithoutDefaults = container.settings.toJSON( { remove: [ 'default' ] } );
+
+		const entries = Object.entries( settingsWithoutDefaults )
+			// Remove controls that are not exist.
+			.filter( ( [ settingName ] ) => !! controls[ settingName ] );
+
+		return Object.fromEntries( entries );
+	}
+
+	extractSpecialSettings( type, container ) {
+		const controls = container.settings.controls,
+			localSettings = container.settings.toJSON( { remove: [ 'default' ] } ),
+			specialSettings = localSettings?.[ type ] || {};
+
+		const entries = Object.entries( specialSettings )
+			// Remove controls that have local value.
+			.filter( ( [ settingsName ] ) => ! Object.hasOwn( localSettings, settingsName ) )
```

Comment:
I think it's better to use `Object.prototype.hasOwnProperty`, because using `Object.hasOwn` will include a lot of polyfills from core-js and will increase the build size (well... we need to check that to make sure)

EDIT: According to [this](https://babeljs.io/repl/#?browsers=last%201%20Android%20versions%2Clast%201%20ChromeAndroid%20versions%2Clast%202%20Chrome%20versions%2Clast%202%20Firefox%20versions%2CSafari%20%3E%3D%2014%2CiOS%20%3E%3D%2014%2C%20last%202%20Edge%20versions%2C%20last%202%20Opera%20versions&build=&builtIns=usage&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBAtgTwPICMBWMC8MDeAoASAEMAuGARgBo8BfAbjz1EliKxlTQFNgoA6ABZEISAO5gAFPGTpKMAOQQQcLgFoA1lwTyYASiA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-2&prettier=false&targets=&version=7.19.5&externalPlugins=%40babel%2Fplugin-transform-runtime%407.19.1%2C%40babel%2Fplugin-transform-modules-commonjs%407.18.6&assumptions=%7B%7D) (I configured it similar to our babel) it shouldn't be that big, but IDK, worth checking 🤷

Reply:
Done